src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -214,6 +253,62 @@ private ProcessPublishElementResult processPublishElements(Element doc) throws X
         return new ProcessPublishElementResult(objects, collisionCount.get());
     }
 
+    private Instant getTimestampForObject(final String objectUri, final byte[] decoded, Instant lastModified) {
+        final RepositoryObjectType objectType = RepositoryObjectType.parse(objectUri);
+        try {
+            return switch (objectType) {
+                case Manifest -> {
```

Comment:
I'm curious if the generic signed object parser may be faster. And while I still think it has holes, by now I think using signing time is definitely better (because notBefore is constant between objects when a repo does not use single-use EE certificates).

From [https://www.ietf.org/archive/id/draft-spaghetti-sidrops-cms-signing-time-00.txt](draft-spaghetti-sidrops-cms-signing-time-00.txt)
```
2.1.  Guidance for Publishers

   When serializing RPKI Signed Objects to a filesystem hierarchy for
   RSYNC consumption, the mod-time of the file containing the Signed
   Object MUST be set to the CMS signing-time contained within the
   Signed Object.
```

Reply:
(see a long back and forth on this on SIDROPS for background)

Reply:
Well, suggest any change here :)
I don't think it really matters which timestamp to use exactly, 
it just needs to stay globally the same for the same hash, so
anything would work.

Reply:
Turns out we do not have a generic signed object parser that offers what I wanted, see https://github.com/RIPE-NCC/rsyncit/pull/6/commits/3e76913d82cc31d5e81c36553fd9307eb1c10e65 for a dirty approach with an in-line implementation.

Reply:
I've changed to signingTime